index.js

```diff
@@ -236,6 +237,122 @@ Peer.prototype.send = function (chunk) {
   self._channel.send(chunk)
 }
 
+/**
+ * Add all tracks within a MediaStream to the connection.
+ * @param {MediaStream} stream
+ * @returns {RTCRtpSender[]}
+ */
+Peer.prototype.addStream = function (stream) {
+  var self = this
+
+  self._debug('addStream()')
+
+  var senders = []
+  stream.getTracks().forEach(function (track) {
+    senders.push(self.addTrack(track, stream))
+  })
+  return senders
+}
+
+/**
+ * Add a MediaStreamTrack to the connection. Returns the RTCRtpSender used to remove the track.
+ * @param {MediaStreamTrack} track
+ * @returns {RTCRtpSender}
+ */
+Peer.prototype.addTrack = function (track, stream) {
+  var self = this
+
+  self._debug('addTrack()')
+
+  return self._pc.addTrack(track, stream)
+}
+
+/**
+ * Add all tracks within a MediaStream to the connection.
+ * @param {MediaStream} stream
+ * @returns {RTCRtpSender[]}
+ */
+Peer.prototype.addStream = function (stream) {
+  var self = this
+
+  self._debug('addStream()')
+
+  var senders = []
+  stream.getTracks().forEach(function (track) {
+    senders.push(self.addTrack(track, stream))
+  })
+  return senders
+}
+
+/**
+ * Remove a RTCRtpSender from the connection.
+ * @param {RTCRtpSender} sender
+ * @returns {RTCRtpSender}
+ */
+Peer.prototype.removeSender = function (sender) {
+  var self = this
+
+  self._debug('removeSender()')
+
+  try {
+    self._pc.removeTrack(sender)
+  } catch (err) {
+    if (err.name === 'NS_ERROR_UNEXPECTED') {
+      self._sendersAwaitingStable.push(sender) // HACK: Firefox must wait until (signalingState === stable) https://bugzilla.mozilla.org/show_bug.cgi?id=1133874
+    } else {
+      self.destroy(err)
+    }
+  }
+}
+Peer.prototype.removeTrack = Peer.prototype.removeSender // for symmetry in the API
+
+/**
+ * Remove an array of RTCRtpSenders from the connection.
+ * @param {RTCRtpSender[]} sender
+ * @param {MediaStream} stream
+ * @returns {RTCRtpSender}
+ */
+Peer.prototype.removeSenders = function (senders) {
+  var self = this
+
+  self._debug('removeSenders()')
+
+  senders.forEach(function (sender) {
+    self.removeSender(sender)
+  })
+}
+Peer.prototype.removeStream = Peer.prototype.removeSenders // for symmetry in the API
+
+/**
+ * Renegotiate the peer connection.
+ */
+Peer.prototype.renegotiate = function () {
+  var self = this
+
+  self._debug('renegotiate()')
+  self._negotiate()
+}
+
+Peer.prototype._negotiate = function () {
+  var self = this
+
+  if (self.initiator) {
+    if (self._isNegotiating) {
+      self._queuedNegotiation = true
+      self._debug('already negotiating, queueing')
+    } else {
+      self._debug('start negotiation')
+      self._createOffer()
+    }
+  } else {
+    self._debug('requesting negotiation from initiator')
+    self.emit('signal', JSON.stringify({ // request initiator to renegotiate
```

Comment:
Sending signaling data over a datachannel would be pretty cool... I think we should look at this when we decide on an API for multiplexed datachannels.

Reply:
I think this is actually ideal, if we do add this via a datachannel it should be a separate api.

Reply:
@feross - This new signaling event will cause an error on older versions of simple-peer (which destroys them.) It's required since only the initiator can safely start renegotiation.